apps/experiments/views/experiment.py

```diff
@@ -1596,3 +1598,134 @@ def trends_data(request, team_slug: str, experiment_id: int):
         return JsonResponse({"error": "Failed to load barchart data", "datasets": []}, status=500)
+
+def _serialize_filter_set(fs: FilterSet) -> dict:
+    return {
+        "id": fs.id,
+        "name": fs.name,
+        "table_type": fs.table_type,
+        "filter_params": fs.filter_params,
+        "is_shared": fs.is_shared,
+        "is_starred": fs.is_starred,
+        "is_default_for_user": fs.is_default_for_user,
+        "is_default_for_team": fs.is_default_for_team,
+    }
+
+@login_and_team_required
+def list_filter_sets(request, team_slug: str, table_type: str):
+    qs = FilterSet.objects.filter(
+        team=request.team,
+        table_type=table_type,
+        is_deleted=False,
+    ).filter(models.Q(user=request.user) | models.Q(is_shared=True))
+
+    data = [_serialize_filter_set(fs) for fs in qs.order_by("-is_starred", "name").all()]
+    return JsonResponse({"results": data})
+
+@login_and_team_required
+@require_POST
+def create_filter_set(request, team_slug: str, table_type: str):
+    payload = json.loads(request.body or b"{}")
+
+    name = payload.get("name", "").strip()
+    filter_params = payload.get("filter_params", {})
+    is_shared = bool(payload.get("is_shared", False))
+    is_starred = bool(payload.get("is_starred", False))
+    is_default_for_user = bool(payload.get("is_default_for_user", False))
+    is_default_for_team = bool(payload.get("is_default_for_team", False))
+
+    if not name:
+        return JsonResponse({"error": "name is required"}, status=400)
+
+    with transaction.atomic():
+        if is_default_for_user:
+            FilterSet.objects.filter(
+                team=request.team, user=request.user, table_type=table_type, is_default_for_user=True
+            ).update(is_default_for_user=False)
+        if is_default_for_team:
+            FilterSet.objects.filter(
+                team=request.team, table_type=table_type, is_default_for_team=True
+            ).update(is_default_for_team=False)
+        fs = FilterSet.objects.create(
+            team=request.team,
+            user=request.user,
+            name=name,
+            table_type=table_type,
+            filter_params=filter_params,
+            is_shared=is_shared,
+            is_starred=is_starred,
+            is_default_for_user=is_default_for_user,
+            is_default_for_team=is_default_for_team,
+        )
+    return JsonResponse({"result": _serialize_filter_set(fs)}, status=201)
+
+
+@login_and_team_required
+def edit_or_delete_filter_set(request, team_slug: str, pk: int):
+    try:
+        fs = FilterSet.objects.get(team=request.team, id=pk, is_deleted=False)
+    except FilterSet.DoesNotExist:
+        return JsonResponse({"error": "Not found"}, status=404)
+
+    is_owner = fs.user == request.user
+    is_team_admin = request.team.membership_set.filter(
+        user=request.user,
+        groups__name__in=["Team Admin", "Super Admin"]
+    ).exists()
```

Comment:
You can use the existing `is_team_admin` prop on membership:
```suggestion
    is_team_admin = request.team_membership.is_team_admin
```